File: 3000/golang/3075. Maximize Happiness of Selected Children.go
func maximumHappinessSum(happiness []int, k int) int64 {
	ans := 0
	sort.Ints(happiness)
	for i := 0; i < k; i++ {
		x := happiness[len(happiness)-i-1] - i
		ans += max(x, 0)
	}
	return int64(ans)
}

File: 3000/swift/3075. Maximize Happiness of Selected Children.swift
class Solution {
    func maximumHappinessSum(_ happiness: [Int], _ k: Int) -> Int {
        var sortHappiness = happiness.sorted(by:>)
        var ans: Int64 = 0
        var i=0
        for _ in 0..<min(k,sortHappiness.count){
            let temp = max(sortHappiness[i]-i,0)
            ans += Int64(temp)
            i += 1
        }
        return Int(ans)
    }
}